interview-questions.md

```diff
@@ -8,44 +8,53 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What does calling super do in a class?
 
-Your answer:
+Your answer: Calling super in a class allows you to use keys that have been defined in a parent class for example if the parent class was animal and it had a attribute of name written into it and later you create an instance of dog instead of having to hard code all of the individual attributes within the animal class you can use super to inherit all of the attributes and apply them to the new object.  
 
-Researched answer:
+Researched answer: Calling the super method in a class allows a developer to access the methods created for the parent class and use them on any of the child classes. They get called on like a regular method call.
 
 2. What is a gem?
 
-Your answer:
+Your answer: a gem is a library similar to a node from react that contains code that has been created for the benefit of the community. If you thought of a specific functionality somebody else has probably already thought of a solution and a gem is how that solution is shared.
 
-Researched answer:
+Researched answer: a gem is a ruby library that contains code that will grant the developer the functionality of the code withiin the gem without having to code it. One example of a gem would be RSPEC the behavior based testing development tool is a gem that can be accessed.
 
 3. What is a relational database? Are there other kinds of databases?
 
-Your answer:
+Your answer: A relational database is comprised linked tables by a specified relationship. It could be a has_many: belongs_to relationship. 
 
-Researched answer:
+Researched answer: A relational database allows for developers to create tables that can be linked to one another based on a primary key. It allows us to collect data and associate items that have some degree of a relationship. The parent table will recieve the primary key and any child table recieves a foreign key that is identical to the primary key to establish a link. 
 
-4. What are primary keys? Why are they important?
+4. What are primary keys? Why are they important? 
 
-Your answer:
+Your answer: Primary keys are used when referencing databases on sql. They will allow models to be associated with one another. It allows for a table to be referenced 
 
-Researched answer:
+Researched answer: The primary key  is  a unique identifier that gets assigned to data as it gets collected within one of the models in the database. It can later be referenced in additional tables to create a relationship with any child tables and have access to the data.
 
 5. What are the HTTP verbs? What is each verb's corresponding CRUD action?
 
-Your answer:
+Your answer: The HTTP verbs are:
+    get: Reads data
```

Comment:
I once told an interviewer I could better show them, and literally went from being stuck to flowing with my super informative answer as I was writing out an example on CRUD and Restful Routes. I am all for this because in the hot seat of an interview, the visual will help you remember what to talk about next. Even if it is not a white boarding section yet I use a piece of paper and do this off camera and talk them through what I understand. These are just words of encouragement. Rant over. Great job Oscar!